qa/integration-tests/src/test/java/io/zeebe/broker/it/health/HealthMonitoringTest.java

```diff
@@ -77,4 +81,21 @@ private void corruptAllSnapshots(final Broker leader) {
                   .forEach(file -> file.delete());
             });
   }
+
+  private boolean isBrokerHealthy(final Broker broker) {
+    final SocketBindingCfg monitoringApi = broker.getConfig().getNetwork().getMonitoringApi();
+    final String baseUrl =
+        String.format("http://%s:%d", monitoringApi.getHost(), monitoringApi.getPort());
+    final String url = baseUrl + "/health";
+
+    try (final CloseableHttpClient client = HttpClients.createDefault()) {
+      final HttpGet request = new HttpGet(url);
+      try (final CloseableHttpResponse response = client.execute(request)) {
+        return response.getStatusLine().getStatusCode() != 503;
+      }
+    } catch (final IOException e) {
+      e.printStackTrace();
+    }
+    return false;
+  }
```

Comment:
```suggestion
  private boolean isBrokerHealthy(final Broker broker) {
    final var monitoringApi = broker.getConfig().getNetwork().getMonitoringApi();
    final var host = monitoringApi.getHost();
    final var port = monitoringApi.getPort();
    final var uri = URI.create(String.format("http://%s:%d/health", host, port));
    final var client = HttpClient.newHttpClient();
    final var request = HttpRequest.newBuilder(uri).build();

    try {
      return client.send(request, BodyHandlers.discarding()).statusCode() == 204;
    } catch (InterruptedException | IOException e) {
      LangUtil.rethrowUnchecked(e);
      return false; // should not happen
    }
  }
```

Java 11 comes with a build in usable `HttpClient` (although still no `URIBuilder` :roll_eyes:) so we don't need the Apache Components dependency here. We can also reuse it for the other test above.

Reply:
:+1: